src/memray/_memray.pyx

```diff
@@ -915,6 +919,10 @@ cdef class FileReader:
                     self._memory_snapshots.push_back(reader.getLatestMemorySnapshot())
                 else:
                     break
+
+        if len(self._memory_snapshots) > max_memory_records:
+            self._memory_snapshot_bucket = len(self._memory_snapshots) // max_memory_records
+            self._memory_snapshots = self._memory_snapshots[::self._memory_snapshot_bucket]
```

Comment:
Why is `self._memory_snapshot_bucket` an instance attribute? It seems like it's only set in this one place, and only used on the next line. Can't it be a local variable?

Also... this math doesn't seem to work. Let's say `max_memory_records == 10_000`, and `len(self._memory_snapshots) == 19_999`, then `len(self._memory_snapshots) // max_memory_records` will be `math.floor(1.9999)`, and so we'll use `1` as the stride and copy every record and still have just under twice as many records as the supposed max.

If we want to use a constant stride, I think we need a `math.ceil` in here:
```suggestion
            stride = math.ceil(len(self._memory_snapshots) / max_memory_records)
            self._memory_snapshots = self._memory_snapshots[::stride]
```

But using a constant stride might not be optimal... If we've got 10,001 snapshots, we'd need to choose between using a stride of 1 (leaving us with 10,001 snapshots, more than the maximum) or using a stride of 2 (leaving us with 5,000 snapshots, only _half_ of the maximum).

An alternative approach would be seeing how many milliseconds elapsed between the timestamp in the first snapshot and the last snapshot, dividing that total number of elapsed milliseconds by our maximum number of snapshots to retain, and keeping only the first snapshot for each value of `(curr_snapshot.time - first_snapshot.time) // granularity`.  That is, conceptually: divide the timeline into 10k buckets, and keep only the first snapshot that falls into each bucket.

Something like:
```py
        cdef vector[_MemorySnapshot] temp_snapshots
        if max_memory_records == 0:
            self._memory_snapshots.resize(0)
        elif len(self._memory_snapshots) > max_memory_records:
            delta = (
                self._memory_snapshots.back().ms_since_epoch
                - self._memory_snapshots.front().ms_since_epoch
            )
            granularity = delta // max_memory_records
            next_snapshot_time = self._memory_snapshots[0].ms_since_epoch

            temp_snapshots.reserve(max_memory_records)
            for snapshot in self._memory_snapshots:
                if snapshot.ms_since_epoch >= next_snapshot_time:
                    temp_snapshots.push_back(snapshot)
                    next_snapshot_time += granularity
```

That compiles and passes the test suite, but I haven't really invested any effort in exercising it thoroughly.

Reply:
Talked offline about this, we will go with the simpler stride-based approach here. The PR was missing some updates that I didn't include after a rebase so that's why the `_memory_snapshot_bucket` was an instance attribute.